backend/uchicago/src/coursesearch_spider.py

```diff
@@ -0,0 +1,128 @@
+import collections
+import re
+import scrapy
+import urllib
+
+from scrapy.selector import Selector
+from scrapy.http import FormRequest
+
+from .course import Course
+from .section import Section
+
+class CourseSearchSpider(scrapy.Spider):
+  name = "coursesearch-spider`"
+  start_urls = [
+      'http://coursesearch.uchicago.edu/',
+  ]
+  COURSE_SEARCH_URL = 'https://coursesearch.uchicago.edu/psc/prdguest/EMPLOYEE/HRMS/c/UC_STUDENT_RECORDS_FL.UC_CLASS_SEARCH_FL.GBL'
+
+  def __init__(self, **kw):
+    super().__init__()
+    self.label = kw.get('label')
+    self.id = kw.get('id')
+
+    self.descriptor_regex = re.compile(r'(?P<id>[A-Z]{4} [0-9]{5})\/(?P<section>[0-9A-Za-z]+) \[(?P<section_id>[0-9]+)\] - (?P<type>[A-Z]+)')
+    
+  def _generate_vars(self, selector):
+    vars = {}
+    for input in selector.css("input[name^='IC']"):
+      vars[input.xpath('@name').extract_first()] = input.xpath('@value').extract_first() 
+    vars['UC_CLSRCH_WRK2_STRM'] = self.id
+    return vars
+  
+  def parse(self, response):
+    vars = self._generate_vars(response)
+    
+    #First, set the session term to the selected term
+    vars['ICAction'] = 'UC_CLSRCH_WRK2_STRM'
+    request =  FormRequest(CourseSearchSpider.COURSE_SEARCH_URL, self._parse_term_data, formdata = vars)
+    yield request
+    
+  def _parse_term_data(self, response):
+    selector = Selector(text = response.body)
+    vars = self._generate_vars(selector)
+    
+    #Run a search on each department sequentially
+    vars['ICAction'] =  'UC_CLSRCH_WRK2_SEARCH_BTN'
+    
+    departments = selector.css("#UC_CLSRCH_WRK2_SUBJECT option").xpath('@value').extract()
+    
+    departments = list(filter(len, departments))
```

Comment:
Inline the selector here:

`departments = list(filter(len, selector.css("#UC_CLSRCH_WRK2_SUBJECT option").xpath('@value').extract()))`